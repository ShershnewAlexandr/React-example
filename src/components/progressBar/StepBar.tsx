import React from 'react';
import { Grid } from '@material-ui/core';
import './ProgressBarStyles.css';

interface StepBarProps {
  icon: string;
  text: string;
  isActive: boolean;
}

function StepBar(props: StepBarProps) {
  const { icon, text, isActive } = props;
  return (
    <Grid item>
      <div className={`icon ${isActive && 'active'}`}>
        <i className="material-icons md-24">{icon}</i>
      </div>
      <div className="step-text">{text}</div>
    </Grid>
  );
}

export default StepBar;
