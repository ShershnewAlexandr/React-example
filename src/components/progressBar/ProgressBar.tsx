import React from 'react';
import { Redirect } from 'react-router-dom';
import { Grid } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import StepBar from './StepBar';
import Loader from '../loaders/Loader';
import { progressBarSteps } from '../../utils/constants';
import { ProgressBarContainerProps } from '../../containers/progressBar/ProgressBarContainer';

function ProgressBar(props: ProgressBarContainerProps) {
  const { t } = useTranslation();
  const {
    step,
    nextStepAction,
    backStepAction,
    progressBarURL,
    location,
    isLoading,
  } = props;
  return (
    <>
      {location.pathname !== progressBarURL && progressBarURL !== '/' && (
        <Redirect to={progressBarURL} />
      )}
      <div className="progress-container">
        <Grid container direction="row" justify="space-around">
          {progressBarSteps.map((current, index) => (
            <StepBar
              key={index}
              icon={current.icon}
              text={t(current.translationKey)}
              isActive={step >= index + 1}
            />
          ))}
          <div className="progress-line-back" />
          <div className={`progress-line progress-line-step${step}`} />
        </Grid>
      </div>
      <Grid
        container
        direction="row"
        justify="space-between"
        alignItems="center"
      >
        <button
          type="button"
          className={isLoading ? 'button disabled' : 'button'}
          onClick={backStepAction}
          disabled={isLoading}
        >
          <Grid container justify="space-between" alignItems="center">
            <Grid item>
              <i className="material-icons back-icon">arrow_back</i>
            </Grid>
            <Grid item>
              <div className="back-but-txt">{t('progressBar.backStep')}</div>
            </Grid>
          </Grid>
        </button>
        <button
          type="button"
          className={
            isLoading ? 'button button-next disabled' : 'button button-next'
          }
          onClick={nextStepAction}
          disabled={isLoading}
        >
          <Grid container justify="space-around" alignItems="center">
            <Grid item>
              <div className="forward-but-txt">{t('progressBar.nextStep')}</div>
            </Grid>
            <Grid item>
              <i className="material-icons forward-icon">arrow_forward</i>
            </Grid>
          </Grid>
        </button>
      </Grid>
      {isLoading && <Loader />}
    </>
  );
}

export default ProgressBar;
