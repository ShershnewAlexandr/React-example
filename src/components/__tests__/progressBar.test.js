import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import '../../setupTests';
import ProgressBar from '../progressBar/ProgressBar';
import StepBar from '../progressBar/StepBar';

jest.mock('../../store/index');
jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: str => str,
  }),
}));

describe('progressBar', () => {
  const progressBarProps = {
    step: 3,
    nextStepAction: jest.fn(),
    backStepAction: jest.fn(),
    progressBarURL: '/',
    location: { pathname: '/' },
  };

  const stepBarProps = {
    text: 'text',
    icon: 'icon',
    isActive: false,
  };

  const progressBar = shallow(
    <ProgressBar {...progressBarProps} isLoading={false} />,
  );
  const progressBarLoading = shallow(
    <ProgressBar {...progressBarProps} isLoading={true} />,
  );
  const stepBar = shallow(<StepBar {...stepBarProps} />);

  it('ProgressBar snapshot', () => {
    expect(toJson(progressBar)).toMatchSnapshot();
  });

  it('ProgressBarLoading snapshot', () => {
    expect(toJson(progressBarLoading)).toMatchSnapshot();
  });

  it('LoginPage snapshot', () => {
    expect(toJson(stepBar)).toMatchSnapshot();
  });
});
