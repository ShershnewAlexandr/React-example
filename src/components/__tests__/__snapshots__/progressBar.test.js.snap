// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`progressBar LoginPage snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  item={true}
>
  <div
    className="icon false"
  >
    <i
      className="material-icons md-24"
    >
      icon
    </i>
  </div>
  <div
    className="step-text"
  >
    text
  </div>
</WithStyles(ForwardRef(Grid))>
`;

exports[`progressBar ProgressBar snapshot 1`] = `
<Fragment>
  <div
    className="progress-container"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      direction="row"
      justify="space-around"
    >
      <StepBar
        icon="face"
        isActive={true}
        key="0"
        text="progressBar.contacts"
      />
      <StepBar
        icon="school"
        isActive={true}
        key="1"
        text="progressBar.education"
      />
      <StepBar
        icon="work"
        isActive={true}
        key="2"
        text="progressBar.experience"
      />
      <StepBar
        icon=""
        isActive={false}
        key="3"
        text=""
      />
      <div
        className="progress-line-back"
      />
      <div
        className="progress-line progress-line-step3"
      />
    </WithStyles(ForwardRef(Grid))>
  </div>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
    direction="row"
    justify="space-between"
  >
    <button
      className="button"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
    >
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        justify="space-between"
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <i
            className="material-icons back-icon"
          >
            arrow_back
          </i>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <div
            className="back-but-txt"
          >
            progressBar.backStep
          </div>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </button>
    <button
      className="button button-next"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
    >
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        justify="space-around"
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <div
            className="forward-but-txt"
          >
            progressBar.nextStep
          </div>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <i
            className="material-icons forward-icon"
          >
            arrow_forward
          </i>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </button>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;

exports[`progressBar ProgressBarLoading snapshot 1`] = `
<Fragment>
  <div
    className="progress-container"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      direction="row"
      justify="space-around"
    >
      <StepBar
        icon="face"
        isActive={true}
        key="0"
        text="progressBar.contacts"
      />
      <StepBar
        icon="school"
        isActive={true}
        key="1"
        text="progressBar.education"
      />
      <StepBar
        icon="work"
        isActive={true}
        key="2"
        text="progressBar.experience"
      />
      <StepBar
        icon=""
        isActive={false}
        key="3"
        text=""
      />
      <div
        className="progress-line-back"
      />
      <div
        className="progress-line progress-line-step3"
      />
    </WithStyles(ForwardRef(Grid))>
  </div>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
    direction="row"
    justify="space-between"
  >
    <button
      className="button disabled"
      disabled={true}
      onClick={[MockFunction]}
      type="button"
    >
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        justify="space-between"
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <i
            className="material-icons back-icon"
          >
            arrow_back
          </i>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <div
            className="back-but-txt"
          >
            progressBar.backStep
          </div>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </button>
    <button
      className="button button-next disabled"
      disabled={true}
      onClick={[MockFunction]}
      type="button"
    >
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        justify="space-around"
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <div
            className="forward-but-txt"
          >
            progressBar.nextStep
          </div>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <i
            className="material-icons forward-icon"
          >
            arrow_forward
          </i>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </button>
  </WithStyles(ForwardRef(Grid))>
  <Loader />
</Fragment>
`;
